### ISSUE TITLE
Cypress installation error

### ISSUE THREAD

**Issue by ehmicky** [jlengstorf/netlify-plugin-visual-diff#1]
It seems like some builds are having issues running [`cypress install`](https://github.com/jlengstorf/netlify-plugin-visual-diff/blob/8ca55e2b06c8b35203747d585e080f9c9de37cf8/index.js#L16):

```
┌───────────────────────────────────────────────────────┐
│ 1. onPreBuild command from netlify-plugin-visual-diff │
└───────────────────────────────────────────────────────┘

no Cypress cache found — installing it now

┌────────────────────────────────────────────────────┐
│ Plugin "netlify-plugin-visual-diff" internal error │
└────────────────────────────────────────────────────┘

Error message
Error: Command failed with ENOENT: cypress install
spawn cypress ENOENT
```

`cypress` is present in `package.json`, then it should be available in `node_modules/.bin`, so I am not sure what might happen. Something that would debug this problem: if the command throws, print `error.stderr` and `error.stdout` (note this requires `stdio` option to be `pipe` or not defined).

**Comment by jlengstorf** [comment]
I followed the lead of `netlify-plugin-cypress` to store the Cypress binary between builds:
https://github.com/jlengstorf/netlify-plugin-visual-diff/blob/8ca55e2b06c8b35203747d585e080f9c9de37cf8/index.js#L94-L102

I can see that this might be causing issues, though (https://github.com/cypress-io/netlify-plugin-cypress/issues/50)

I have no idea why this wouldn't be available, though. this kind of points back to my suspicion that the cache utils are flaky sometimes (see also https://github.com/netlify/build/issues/1225) — I think they sometimes confirm a save, then miss on restoration

I also have _zero_ idea how a fresh install would be missing. that kind of feels upstream? I'm happy to help debug, but I have no idea where to even start (for example, I don't know what the "`stdio` option" refers to — which link in this chain needs that option set?)

**Comment by ehmicky** [comment]
Related too: https://github.com/netlify/build-image-plugins/issues/5

**Comment by ehmicky** [comment]
It looks like the [`utils.cache.restore()`](https://github.com/jlengstorf/netlify-plugin-visual-diff/blob/8ca55e2b06c8b35203747d585e080f9c9de37cf8/index.js#L12) and [`utils.cache.save()`](https://github.com/jlengstorf/netlify-plugin-visual-diff/blob/8ca55e2b06c8b35203747d585e080f9c9de37cf8/index.js#L94) operations might not be needed here.

When `cypress install` runs, it is added to `~/.cache/Cypress`. `~/.cache` is saved and restored to the build cache by default by Netlify buildbot. `cypress install` detects this automatically and becomes a quick noop instead when Cypress has already been installed/cached.

Actually, if users are already using `cypress` (including through `netlify-plugin-cypress`), this will be cached in `~/.cache/Cypress`, so having it cached in `/opt/buildhome/.cache/Cypress` would create duplicate cache (of `580MB`). 

Also, `../../buildhome/.cache/Cypress` would not work when using a `base` directory, since some additional `..` might be needed to reach `/opt/buildhome/`.

On a different topic: `cypress install` itself might not be needed. Unless I'm mistaken, this operation is done on `postinstall` by Cypress. Since `cypress` is a dependency of `netlify-plugin-visual-diff`, I think removing that call altogether should work.

I think it might be interesting to see if the plugin still work after removing the two following things?
  - `utils.cache.restore()` and `utils.cache.save()`
  - `utils.run('cypress', ['install'])`

**Comment by jlengstorf** [comment]
this was fixed in #2 — please let me know if you see anything in the logs for this one!

**Comment by ehmicky** [comment]
Thanks @jlengstorf!
